yare-engine/src/main/java/com/sabre/oss/yare/invoker/java/MethodCallMetadataFactory.java

```diff
@@ -50,6 +51,7 @@
     @SuppressWarnings("unchecked")
     static <T> MethodCallMetadata method(T instance, CatchingInvocation<T> invocation) {
         Class<?> targetType = instance.getClass();
+        validateActionMappingCandidate(targetType);
```

Comment:
I guess that code would be a bit more readable without that static import here.
Q: Do we really need that validator to be static? As far I understood this is the only class it is used.

Reply:
1. The context is static - "static <T> MethodCallMetadata method(..."
2.  MethodCallMetadataValidator holds no state.

Any reason why "private static MethodCallMetadataValidator methodCallMetadataValidator = new MethodCallMetadataValidator();" is better?

Reply:
Cool, I do not have any strong feelings about that. I was rather thinking about instantiation of validator as static property of the MCMFactory as we consider this validator as factory's integral part. Of course we can still talk about potential future cases but... YAGNI